src/allotropy/parsers/revvity_kaleido/kaleido_structure_v3.py

```diff
@@ -0,0 +1,177 @@
+import re
+
+from allotropy.parsers.lines_reader import CsvReader
+from allotropy.parsers.revvity_kaleido.kaleido_structure import (
+    AnalysisResult,
+    BackgroundInfo,
+    Data,
+    MeasurementElement,
+    MeasurementInfo,
+    Measurements,
+    Platemap,
+    PlateType,
+    Results,
+)
+from allotropy.parsers.utils.values import assert_not_none
+
+
+def create_background_info(reader: CsvReader) -> BackgroundInfo:
+    assert_not_none(
+        reader.pop_if_match("^EnSight Results from"),
+        msg="Unable to find EnSight section.",
+    )
+
+    reader.drop_until("^Result for")
+
+    line = assert_not_none(
+        reader.drop_until_inclusive("^Result for.(.+) 1"),
+        msg="Unable to find background information.",
+    )
+
+    experiment_type = assert_not_none(
+        re.match("^Result for.(.+) 1", line),
+        msg="Unable to find experiment type from background information section.",
+    ).group(1)
+
+    return BackgroundInfo(experiment_type)
+
+
+def create_results(reader: CsvReader) -> Results:
+    assert_not_none(
+        reader.pop_if_match("^Barcode"),
+        msg="Unable to find barcode indicator.",
+    )
+
+    barcode, *_ = assert_not_none(
+        reader.pop_if_match("^.+,"),
+        msg="Unable to find barcode value.",
+    ).split(",", maxsplit=1)
+
+    results = assert_not_none(
+        reader.pop_csv_block_as_df(header=0, index_col=0),
+        msg="Unable to find results table.",
+    )
+
+    for column in results:
+        if str(column).startswith("Unnamed"):
+            results = results.drop(columns=column)
+
+    return Results(
+        barcode=barcode,
+        results=results,
+    )
+
+
+def create_analysis_results(reader: CsvReader) -> list[AnalysisResult]:
```

Comment:
It looks to me like this is identical to v2, move to common?

Reply:
There is a difference in the section titles, one is Measurement Basic Information, the other is Measurement Information.

Reply:
Ah, I did not spot that, thanks 👍